griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,270 @@
+
+    owner_email: str = field(default=None)
+
+    @activity(
+        config={
+            "description": "Can be used to create a new Google Doc in Drive",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_name",
+                        description="Name of the file to be created."
+                    ): str
+                }
+            ),
+        }
+    )
+    def create_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_name = params["file_name"]
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().create(body={"title": file_name}).execute()
+            return InfoArtifact(f"Google Doc '{file_name}' created with ID: {doc['documentId']}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to append text to a Google Doc",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Destination file path on Google Drive in the POSIX format. "
+                                    "For example, 'foo/bar/baz.txt'"
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be appended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def append_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+                content = doc["body"]["content"]
+                last_text = content[-1]["paragraph"]["elements"][-1]["textRun"]["content"]
+                append_index = content[-1]["endIndex"]
+                if last_text.endswith("\n"):
+                    append_index -= 1
+
+                requests = [{"insertText": {"location": {"index": append_index}, "text": text}}]
+
+                doc = service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("text appended successfully")
+            else:
+                return ErrorArtifact(f"error appending to Google Doc, file not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error appending text to Google Doc with path {file_path}: {e}")
```

Comment:
`ErrorArtifact` content seems to be different even though you are communicating the same message. Should be the same.

Reply:
the above error is for if the file is not found, but this error is for if some other error happens that does not allow appending to the file at the file_path.